02-microarray/gene-id-annotation_microarray_01_ensembl.Rmd

```diff
@@ -0,0 +1,368 @@
+  dplyr::filter(gene_symbol_count > 1)
+
+# Let's look at the first 6 rows of our `multi_mapped` object
+head(multi_mapped)
+```
+
+Looks like we have some cases where 3 gene symbols mapped to a single Ensembl ID.
+We have a total of 130 out of 17984 Ensembl IDs with multiple mappings to gene symbols.
+We are not too worried about the 130 IDs with multiple mappings, so we will filter them out for the purpose of having 1:1 mappings for our downstream analysis.
+
+## Map Ensembl IDs to gene symbols -- filtering out multi mappings
+
+The next code chunk will rerun the `mapIds()` function, this time supplying the `"filter"` option to the `multiVals` argument.
+This will remove all instances of multiple mappings and return a list of only the gene identifiers and symbols that had 1:1 mapping.
+Use `?mapIds` to see more options or strategies.
+
+```{r}
+# Map ensembl IDs to their associated gene symbols
+filtered_mapped_df <- mapIds(
+  org.Mm.eg.db, # Replace with annotation package for the organism relevant to your data
+  keys = df$Gene,
+  column = "SYMBOL", # Replace with the type of gene identifiers you would like to map to
+  keytype = "ENSEMBL", # Replace with the type of gene identifiers in your data
+  multiVals = "filter"
+) %>%
+  # Turn our list into a data frame
+  reshape2::melt() %>%
+  # Now let's give the `value` column a name that is more relevant to the values it holds
+  dplyr::rename("Symbol" = "value")
+```
+
+Let's copy the same code we used above to see if we have any multi mappings in our new data frame.
+
+```{r}
+filtered_mapped_df %>%
+  # First, we need to grab the Ensembl IDs from the rownames
+  tibble::rownames_to_column("Gene") %>%
+  # Let's group by the Ensembl ID's in the `L1` column
+  dplyr::group_by(Gene) %>%
+  # Create a new variable containing the number of symbols mapped to each ID
+  dplyr::mutate(gene_symbol_count = dplyr::n()) %>%
+  # Arrange by the genes with the highest number of symbols mapped
+  dplyr::arrange(desc(gene_symbol_count)) %>%
+  # Filter to include only the rows with multi mappings
+  dplyr::filter(gene_symbol_count > 1)
+```
+
+This new new `filtered_mapped_df` object is empty, as expected, since it has been filtered to include only 1:1 mappings.
+
+Now, let's write our filtered and mapped results to file!
+
+## Write mapped results to file
+
+```{r Write to file}
+# Write mapped and annotated data frame to output file
+readr::write_tsv(filtered_mapped_df, file.path(
+  results_dir,
+  "GSE13490_Gene_Symbols.tsv" # Replace with a relevant output file name
+))
+```
+
+
+# Resources for further learning
+
+- Marc Carlson has prepared a nice [Introduction to Bioconductor Annotation Packages](https://bioconductor.org/packages/release/bioc/vignettes/AnnotationDbi/inst/doc/IntroToAnnotationPackages.pdf) [@Carlson2020]
+- [More on Annotation mapping functions](https://www.bioconductor.org/packages/release/bioc/vignettes/AnnotationFuncs/inst/doc/AnnotationFuncsUserguide.pdf) [@Edwards2020]
```

Comment:
I wouldn't share this alternative annotation package's user guide, I find it more confusing than helpful. Easier just to share AnnotationDbi user guide tutorials as opposed to sending our users to something else that they would have to learn from scratch.

Reply:
Gotcha, that makes sense. I removed this reference but haven't found any additional user guide tutorials thus far, do you have any suggestions or personal favorites for this particular case?

Reply:
I don't think I have anything else. We can add more if we find something.